### PR TITLE
Block timestamp bugfix

### PR DESCRIPTION
**Pull request by mikiw**
## Development related changes

- Add of new contract and test to test timestamp call

## Checklist:

- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/clippy_check.sh`
- [x] Performed code self-review
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Documented the changes
- [x] Linked the issues which this PR resolves
- [x] Checked the TODO section in README.md if this PR resolves it
- [x] Updated the tests
- [x] All tests are passing - `cargo test`

### CODE REVIEW

**Comment by mikiw** [comment]
related to https://github.com/starkware-libs/blockifier/issues/1145

File: crates/starknet/src/starknet/mod.rs [hunk 1]
```diff
@@ -322,33 +323,37 @@ impl Starknet {
         use starknet_api::hash::StarkHash;
         use starknet_api::{contract_address, patricia_key};
```
[the remark this answers is not in the document]
Comment (mikiw) [reply crates/starknet/src/starknet/mod.rs:324]: I think that it's there intentionally right @marioiordanov ?

File: crates/starknet/src/starknet/mod.rs [hunk 2]
```diff
@@ -322,33 +323,37 @@ impl Starknet {
...
+                ("ecdsa_builtin".to_string(), N_STEPS_FEE_WEIGHT * 2048.0),
+                ("bitwise_builtin".to_string(), N_STEPS_FEE_WEIGHT * 64.0),
+                ("ec_op_builtin".to_string(), N_STEPS_FEE_WEIGHT * 1024.0),
+                ("poseidon_builtin".to_string(), N_STEPS_FEE_WEIGHT * 32.0),
+                ("segment_arena_builtin".to_string(), N_STEPS_FEE_WEIGHT * 10.0),
+                ("keccak_builtin".to_string(), N_STEPS_FEE_WEIGHT * 2048.0), // 2**11
+            ])),
+            gas_prices: blockifier::block_context::GasPrices {
+                eth_l1_gas_price: gas_price as u128,
+                strk_l1_gas_price: gas_price as u128,
```
[the remark this answers is not in the document]
Comment (mikiw) [reply crates/starknet/src/starknet/mod.rs:351]: they have same value in `create_for_testing()`

File: crates/starknet/src/starknet/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (marioiordanov) [reply crates/starknet/src/starknet/mod.rs:324]: yes, to be able to use the contract_address! macro. Some compilation errors arised if those imports were out of the scope of the method

File: crates/starknet/src/starknet/mod.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (FabijanC) [reply crates/starknet/src/starknet/mod.rs:351]: ok, let's hope we'll remember to change it in the future; resolving

**Review by FabijanC — APPROVED** [review]

**Pull request merged**